### PR TITLE
[Merged by Bors] - feat: `casesm` / `constructorm` tactic

### PR DESCRIPTION
**Pull request by digama0**

### CODE REVIEW

File: Mathlib/Tactic/CasesM.lean [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    g.withContext do
+      for ldecl in ← getLCtx do
+        if ← matcher ldecl.type then
+          let mut acc := acc
+          let subgoals ← if allowSplit then
+            g.cases ldecl.fvarId
+          else
+            let s ← saveState
+            let subgoals ← g.cases ldecl.fvarId
+            if subgoals.size > 1 then
```
Comment (semorrison) [inline Mathlib/Tactic/CasesM.lean:36]: I think you're missing an `if allowSplit` here.

Comment (digama0) [reply Mathlib/Tactic/CasesM.lean:36]: this is in the `else` branch of an `if allowSplit`

**Comment by semorrison** [comment]
LGTM, besides the merge conflict.

**Comment by digama0** [comment]
bors merge

**Pull request closed without merging**